base/complex.jl

```diff
@@ -366,11 +366,15 @@ function /(a::Complex{T}, b::Complex{T}) where T<:Real
     end
 end
 
-inv(z::Complex{<:Union{Float16,Float32}}) =
-    oftype(z, inv(widen(z)))
-
-/(z::Complex{T}, w::Complex{T}) where {T<:Union{Float16,Float32}} =
-    oftype(z, widen(z)*inv(widen(w)))
+function /(z::Complex{T}, w::Complex{T}) where {T<:Union{Float16,Float32}}
+    c, d = reim(widen(w))
+    a, b = reim(widen(z))
+    (isinf(c) | isinf(d)) && return complex(copysign(zero(T), c), flipsign(zero(T), d)) * z
```

Comment:
What's the sign of a complex number?  `copysign(zero(T), c)` doesn't make much sense (and it's where this is failing)